### PR TITLE
HS-1140 Allow matching multiple monitoring policies for alerts

### PR DESCRIPTION
**Pull request by BenD10**
## Description
Alerts may have multiple monitoring policies associated with them, so the notification service needs to honor that.

## Jira link(s)
- https://issues.opennms.org/browse/HS-1140

## Checklist
* [x] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [x] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [x] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts

### CODE REVIEW

**Review by vserrago — COMMENTED** [review]
Looks good to me, but I'll defer to one of the horizon stream maintainers.

**Review by yangli2014 — APPROVED** [review]
LGTM

File: notifications/src/main/java/org/opennms/horizon/notifications/service/NotificationService.java [hunk 1]
```diff
@@ -53,12 +54,12 @@ public class NotificationService {
...
     public void postNotification(Alert alert) {
-        Optional<MonitoringPolicy> dbPolicy = monitoringPolicyRepository.findByTenantIdAndId(
+        List<MonitoringPolicy> dbPolicies = monitoringPolicyRepository.findByTenantIdAndIdIn(
             alert.getTenantId(),
-            alert.getMonitoringPolicyId()
+            alert.getMonitoringPolicyIdList()
         );
 
-        dbPolicy.ifPresentOrElse(policy -> {
+        dbPolicies.forEach(policy -> {
```
Comment (james8hut) [inline notifications/src/main/java/org/opennms/horizon/notifications/service/NotificationService.java:62]: Question on this: If we have multiple monitoring policies which each have the notifyByPagerDuty flag set, then we'll get multiple calls to PagerDuty for the same alert. Is this really what we want here?

Comment (BenD10) [reply notifications/src/main/java/org/opennms/horizon/notifications/service/NotificationService.java:62]: That's a good point, maybe it has to be more a deduplication, if PD is enabled in at least one policy, send it, if email is enabled in atleast one of the policies, send it, etc.

Comment (james8hut) [reply notifications/src/main/java/org/opennms/horizon/notifications/service/NotificationService.java:62]: As I'm just back from vacation and at a loose end, I'll create a story for this and do it. :-)

**Pull request merged**